processing/src/main/java/org/apache/druid/query/aggregation/BufferAggregator.java

```diff
@@ -158,7 +158,9 @@ default double getDouble(ByteBuffer buf, int position)
   }
 
   /**
-   * Release any resources used by the aggregator
+   * Release any resources used by the aggregator. The aggregator may be reused after this call. This call would be
```

Comment:
Should we state this change in release notes, incase somebody is doing a proper `close` behavior in their custom aggregator? 
Also, what would be the correct thing to do if an aggregator has meaningful and different behaviours for both reset and close call?

Reply:
Yes, I think it makes sense to mention this in the release notes. I added a label.

>  Also, what would be the correct thing to do if an aggregator has meaningful and different behaviours for both reset and close call?

I'm not sure what you mean by this. Currently there is only one method, `close()`. In that method, the aggregator should release everything it is holding. Afterwards, if some other method like `init` is called, the aggregator should re-acquire or re-set-up anything it needs.

Fwiw, I did consider keeping the `close()` contract such that the aggregator is really closed and won't be reused afterwards, but didn't stick with that for two reasons:

- aggregators in core didn't need the contract to be that tight
- tightening the contract like that would require re-calling `factorizeBuffered` or `factorizeVector` each time the grouper is reset, which seemed wasteful.

Reply:
Yeah my intent here was that if an aggregator instance is reused after `close`, it should be fresh, so should start again with `init`. I updated the comments to clarify that.

Reply:
Thanks for the update, lgtm